test/jdk/java/awt/Clipboard/LostOwnershipChainTest/SystemClipboard2ProcTest.java

```diff
@@ -0,0 +1,170 @@
+ * @library /test/lib
+ * @run main SystemClipboard2ProcTest
+*/
+
+import java.awt.Toolkit;
+import java.awt.datatransfer.Clipboard;
+import java.awt.datatransfer.ClipboardOwner;
+import java.awt.datatransfer.DataFlavor;
+import java.awt.datatransfer.StringSelection;
+import java.awt.datatransfer.Transferable;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.process.ProcessTools;
+
+public class SystemClipboard2ProcTest {
+
+    public static void main(String[] args) throws Exception {
+        SystemClipboardOwner.run();
+
+        if (SystemClipboardOwner.failed) {
+            throw new RuntimeException("test failed: can not get actual " +
+                    "contents of the clipboard or set owner of the clipboard");
+        } else {
+            System.err.println("test passed");
+        }
+    }
+}
+
+class SystemClipboardOwner implements ClipboardOwner {
+    static volatile boolean failed;
+
+    private static final Object LOCK = new Object();
+
+    private static final int CHAIN_LENGTH = 5;
+    private final static Clipboard clipboard =
+        Toolkit.getDefaultToolkit().getSystemClipboard();
+
+    private int m, id;
+
+    public SystemClipboardOwner(int m) { this.m = m; id = m; }
+
+    public void lostOwnership(Clipboard cb, Transferable contents) {
+        System.err.println(id + " lost clipboard ownership");
+
+        Transferable t = getClipboardContents(cb, null);
+        // for test passing if t.getTransferData() will throw an exception
+        String msg = "" + (m + 1);
+        try {
+            msg = (String)t.getTransferData(DataFlavor.stringFlavor);
+        } catch (Exception e) {
+            System.err.println(id + " can't getTransferData: " + e);
+        }
+        System.err.println(id + " Clipboard.getContents(): " + msg);
+        if (!msg.equals("" + (m + 1))) {
+            failed = true;
+            System.err.println("Clipboard.getContents() returned incorrect contents!");
+        }
+
+        m += 2;
+        if (m <= CHAIN_LENGTH) {
+            System.err.println(id + " Clipboard.setContents(): " + m);
+            setClipboardContents(cb, new StringSelection(m + ""), this);
+        }
+        if (m >= CHAIN_LENGTH) {
+            synchronized (LOCK) {
+                LOCK.notifyAll();
+            }
+        }
+    }
+
+    public static void run() throws Exception {
+        SystemClipboardOwner cbo1 = new SystemClipboardOwner(0);
+        System.err.println(cbo1.m + " Clipboard.setContents(): " + cbo1.m);
+        setClipboardContents(clipboard, new StringSelection(cbo1.m + ""),
+                             cbo1);
+
+        ProcessBuilder pb = ProcessTools
+                .createTestJavaProcessBuilder(SystemClipboardOwner.class.getName());
+
+        Process process = ProcessTools.startProcess("Child", pb);
+        OutputAnalyzer outputAnalyzer = new OutputAnalyzer(process);
```

Comment:
OutputAnalyzer is not really needed here, as output is printed by `ProcessTools`, exit value can be received from `process.exitValue()` , however it brings the `shouldHaveExitValue` method and prints some useful info like:

```
Command line: [.../jdk/bin/java -cp ... -ea SystemClipboardOwner ]
[Child]:.../jdk/bin/java -cp ... -ea SystemClipboardOwner
[2025-04-23T20:51:22.569840505Z] Gathering output for process 165156
[2025-04-23T20:51:23.984714408Z] Waiting for completion for process 165156
[2025-04-23T20:51:23.984831108Z] Waiting for completion finished for process 165156
```

So I decided to use it.